x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -160,17 +165,40 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                   return { createdAlerts: [], errors: {}, alertsWereTruncated };
                 }
 
-                return {
-                  createdAlerts: augmentedAlerts
-                    .map((alert, idx) => {
-                      const responseItem = response.body.items[idx].create;
-                      return {
-                        _id: responseItem?._id ?? '',
-                        _index: responseItem?._index ?? '',
-                        ...alert._source,
-                      };
+                const createdAlerts = augmentedAlerts
+                  .map((alert, idx) => {
+                    const responseItem = response.body.items[idx].create;
+                    return {
+                      _id: responseItem?._id ?? '',
+                      _index: responseItem?._index ?? '',
+                      ...alert._source,
+                    };
+                  })
+                  .filter((_, idx) => response.body.items[idx].create?.status === 201)
+                  // Security solution's EQL rule consists of building block alerts which should be filtered out.
+                  // Building block alerts have additional "kibana.alert.group.index" attribute which is absent for the root alert.
+                  .filter((alert) => !Object.keys(alert).includes(ALERT_GROUP_INDEX));
+
+                createdAlerts.forEach((alert) =>
+                  options.services.alertFactory
+                    .create(alert._id)
+                    .scheduleActions(type.defaultActionGroupId, {
+                      rule: mapKeys(snakeCase, {
+                        ...options.params,
+                        name: options.rule.name,
+                        id: options.rule.id,
+                      }),
+                      results_link: type.getViewInAppRelativeUrl?.({
+                        rule: options.rule,
```

Comment:
it looks like `options.rule` does not have `params` property which is used within `getViewInAppRelativeUrl`.It leads to creating invalid `result_link`.

It might be the reason I can't see it when using summary alert

Just few lines above, params are getting merged into rule

```
  rule: mapKeys(snakeCase, {
                        ...options.params,
                        name: options.rule.name,
                        id: options.rule.id,
```

cc: @marshallmain

Reply:
Here is the code which builds the results URL for summary actions:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/task_runner/execution_handler.ts#L432

First thing framework does, is checking whether `this.taskRunnerContext.kibanaBaseUrl` is set.